Xamarin.Forms.Platform.iOS/Renderers/EntryRenderer.cs

```diff
@@ -338,19 +362,50 @@ void UpdateCursorSelection()
 			if (_selectedTextRangeIsUpdating || control == null || Element == null)
 				return;
 
-			if (Element.IsSet(Entry.CursorPositionProperty) || Element.IsSet(Entry.SelectionLengthProperty)) {
+			var currentSelection = control.SelectedTextRange;
+
+			if (_defaultCursorStartPosition == null)
+				_defaultCursorStartPosition = currentSelection.Start;
+
+			if (_defaultCursorEndPosition == null)
+				_defaultCursorEndPosition = currentSelection.End;
+
+			// If this is run from the ctor, the control is likely too early in its lifecycle to be first responder yet. 
+			// Anything done here will have no effect, so we'll skip this work until later.
+			// We'll try again when the control does become first responder later OnEditingBegan
+			if (control.BecomeFirstResponder())
+			{
+				int cursorPosition = Element.CursorPosition;
+
+				UITextPosition start;
+				if (Element.IsSet(Entry.CursorPositionProperty))
+					start = control.GetPosition(control.BeginningOfDocument, cursorPosition);
+				else
+					start = _defaultCursorStartPosition;
+
+				int startOffset = (int)control.GetOffsetFromPosition(control.BeginningOfDocument, start); 
+
+				UITextPosition end;
+				if (Element.IsSet(Entry.SelectionLengthProperty))
+					end = control.GetPosition(start, Math.Max(startOffset, Math.Min(control.Text.Length - cursorPosition, Element.SelectionLength)));
+				else
+					end = _defaultCursorEndPosition;
```

Comment:
These defaults look to be forcing the cursor to position zero if the cursor position itself has never been set.
![ee5eea65-6327-4f9b-a5de-bdd27a011fa7](https://user-images.githubusercontent.com/5375137/44063801-69253b60-9f1f-11e8-9f14-63b6241c8d91.GIF)

Initially clicking the second box in the gif causes the cursor to always go to zero. 
If you remove *control.SelectedTextRange = control.GetTextRange(start, end);*  below you'll see that the initial click on the second box goes to the end of the box

on UWP and Android the cursor goes to the end fine